src/plugins/es_ui_shared/docs/docs/form_lib/examples/react_to_changes.md

```diff
@@ -0,0 +1,189 @@
+
+  return (
+    <Form form={form}>
+      <FormFields />
+    </Form>
+  );
+};
+```
+
+## Listen to specific form fields changes
+
+In some cases you only want to listen to some field change and don't want to trigger a re-render of your component for every field value change. You can specify a **watch** (`string | string[]`) parameter for that.
+
+```js
+export const ReactToSpecificFields = () => {
+  const { form } = useForm();
+  const [{ showAddress }] = useFormData({ form, watch: 'showAddress' });
+
+  return (
+    <Form form={form}>
+      {/* Changing the "name" field won't trigger a re-render */}
+      <UseField path="name" config={{ label: 'First name' }} component={TextField} />
+
+      <UseField
+        path="showAddress"
+        config={{ defaultValue: false, label: 'Show address' }}
+        component={ToggleField}
+      />
+      {showAddress && (
+        <>
+          <p>800 W El Camino Real #350</p>
+        </>
+      )}
+    </Form>
+  );
+};
+```
+
+## Using the `onChange` handler
+
+Sometimes the good old `onChange` handler is all you need to react to a form field value change (instead of reading the form data and add a `useEffect` to react to it).
+
+```js
+export const OnChangeHandler = () => {
+  const { form } = useForm();
+
+  const onNameChange = (value: string) => {
+    console.log(value);
+  };
+
+  return (
+    <Form form={form}>
+      <UseField
+        path="name"
+        config={{ label: 'Name' }}
+        component={TextField}
+        onChange={onNameChange}
+      />
+    </Form>
+  );
+};
+```
+
+## Forward the form state to a parent component
+
+If your UX requires to submit the form in a parent component (e.g. because that's where your submit button is located), you will need a way to access the form validity and the form data outside your form component. Unless your parent component needs to be aware of every field value change in the form (which should rarely be needed), you don't want to use the `useFormData()` hook and forward the data from there. This would create unnecessary re-renders. Instead it is better to forward the `getFormData()` handler on the form.
+
+```js
+interface MyForm {
+  name: string;
+}
+
+interface FormState {
+  isValid: boolean | undefined;
+  validate(): Promise<boolean>;
+  getData(): MyForm;
+}
+
+const schema: FormSchema<MyForm> = {
+  name: {
+    validations: [
+      {
+        validator: ({ value }) => {
+          if (value === 'John') {
+            return { message: `The username "John" already exists` };
+          }
+        },
+      },
+    ],
+  },
+};
+
+interface Props {
+  defaultValue: MyForm;
+  onChange(formState: FormState): void;
+}
+
+const MyForm = ({ defaultValue, onChange }: Props) => {
+  const { form } = useForm<MyForm>({ defaultValue, schema });
+  const { isValid, validate, getFormData } = form;
+
+  useEffect(() => {
+    onChange({ isValid, validate, getData: getFormData });
+  }, [onChange, isValid, validate, getFormData]);
```

Comment:
Just to clarify, `getFormData` is a stable reference and that is how we do not trigger a re-render on each form change right? Might be cool to have a code comment to that effect here if that is correct.

Reply:
That's correct. I'll add a comment to indicate it, thanks for the suggestion! 👍